app/stores/SchoolStore.js

```diff
@@ -170,8 +46,110 @@ SchoolStore.dispatchToken = AppDispatcher.register(function(payload) {
   }
 });
 
+function getBeginAndEndYear(school) {
+  const currentName = _.first(school.names);
+  const oldestName = _.last(school.names);
+  return {
+    beginYear: oldestName ? oldestName.begin_year : null,
+    endYear: currentName ? currentName.end_year : null
+  };
+}
+
+function getFetchingData() {
+  return _fetchingData;
+}
+
+function getMainBuilding(school) {
+  return _.first(school.buildings) || {};
+}
+
+function getMainBuildingInYear(schoolId, year) {
+  const school = _schools[schoolId];
+  if (_.isEmpty(school)) {
+    return {};
+  }
+
+  if (!year) {
+    return getMainBuilding(school);
+  }
+
+  const mainBuilding = _.find(school.buildings, function(building) {
+    return building.begin_year <= year;
+  });
+  return mainBuilding || {};
+}
+
+function getMainName(school) {
+  return _.first(school.names) || {};
+}
+
+function getSchoolDetails(school) {
+  return {
+    schoolNames: school.names,
+    buildings: school.buildings
+  };
+}
+
+function getSchools(schoolIds) {
+  return _.map(schoolIds, function(id) {
+    const school = _schools[id];
+    if (_.isEmpty(school)) {
+      return {};
+    }
+    let addresses = [];
+    const mainBuilding = getMainBuilding(school);
+    if (mainBuilding.building && mainBuilding.building.addresses) {
+      addresses = mainBuilding.building.addresses;
+    }
+    const address = (
+      addresses.length ?
+      `${addresses[0].street_name_fi}\, ${addresses[0].municipality_fi}` :
+      ''
+    );
+    return {
+      id: id,
+      name: getMainName(school).official_name,
+      address: address
+    };
+  }, this);
+}
+
+function getSchoolYearDetails(schoolId, year) {
+  const school = _schools[schoolId];
+  if (_.isEmpty(school)) {
+    return {};
+  }
+  year = year || new Date().getFullYear();
+  const schoolName = _.find(school.names, function(name) {
+    return name.begin_year <= year;
+  });
+  const schoolBuilding = _.find(school.buildings, function(building) {
+    return building.begin_year <= year;
+  });
+  return {
+    schoolName: schoolName || {},
+    building: schoolBuilding || {}
+  };
+}
+
+function hasSchool(schoolId) {
+  return typeof _schools[schoolId] !== 'undefined';
+}
+
+function _getSchoolByIdWrapper(func, defaultValue) {
+  return function(schoolId) {
+    defaultValue = defaultValue ? defaultValue : [];
+    const school = _schools[schoolId];
+    return _.isEmpty(school) ? defaultValue : func(school);
+  };
+}
```

Comment:
It would be nice to receive a school object in all get functions instead of the SchoolId. If we will still use this wrapper, maybe we could modify it a bit so it can be used in functions that need more arguments than just the schoolId.

In order to do that we can use the arguments object: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments . arguments array could be copied into another array ( so it's not changed) shifting from it the schoolId and unshifting the school object if founded. After that you can call the callback function using apply.